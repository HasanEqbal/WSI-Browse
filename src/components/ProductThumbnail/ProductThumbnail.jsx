import React, { useState } from 'react';
import Parser from 'html-react-parser';
import { Link } from 'react-router-dom';
import './ProductThumbnail.scss';

export default function ProductThumbnail({ product }) {
  const {
    name,
    id,
    price,
    priceRange,
  } = product;

  let regularHigh
  let regularLow
  let sellingHigh
  let sellingLow

  if (priceRange) {
    if (priceRange.regular) {
      regularHigh = priceRange.regular.high;
      regularLow = priceRange.regular.low;
    }

    if (priceRange.selling) {
      sellingHigh = priceRange.selling.high;
      sellingLow = priceRange.selling.low;
    }
  }

  let regularPrice
  let sellingPrice

  if (price) {
    regularPrice = price.regular
    if (price.selling) {
      sellingPrice = price.selling
    }
  }

  let image = product.thumbnail.href;

  return (
    <React.Fragment>
      <div className="cell thumbnail">
        <div className="product-card" >
          <div className="product-card-thumbnail">
            <Link className="product-main-image" to={`/shop/${id}`}><img src={image} alt={name} /></Link>
          </div>
          <h6 className="product-card-title">
            <Link className="product-name"to={`/shop/${id}`}>{Parser(name)}</Link></h6>
          <div>
            {regularHigh ?
              <span className="product-card-old-price">
                Original Price: ${regularLow}-${regularHigh}
              </span>
              : regularPrice ?
                <span className="product-card-regular-price">
                  Original Price: ${regularPrice}
                </span> : ""
            }
          </div>
          <div>
            <span className="product-card-sale-price">
              Sale Price: ${sellingHigh
                ? `${sellingLow} - $${sellingHigh}`
                : sellingPrice
              }
            </span>
          </div>
        </div>
      </div>
    </React.Fragment >
  )
}

