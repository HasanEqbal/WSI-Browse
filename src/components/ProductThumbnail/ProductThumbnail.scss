
.product-card {
  background-color: #fefefe;
  padding: 1rem;
  margin-bottom: 1.5rem;
  margin-right: .8rem;
  border: #D9524F;
}

.product-card-thumbnail {
  display: inline;
  position: relative;
  background-color: #f3f3f3;
}

.product-card-thumbnail:hover{
  border: #D9524F;
}
.product-card-title {
  font-family: 'Open sans', sans-serif;
  color: #082a4c;
  font-size: 0.9rem;
  font-weight: 600;
  margin-top: 1rem;
  height: 4rem;
  margin-bottom: 0;
}


.product-card-sale-price {
  color: #D9524F;
  display: inline-block;
  font-size: 0.75rem;
  font-weight: 600;
  margin-top: 0.8rem;
  margin-left: 0.2rem;
  text-align: center;
}

.product-card-old-price {
  color: darken(#082a4c, 30%);
  display: inline-block;
  font-size: 0.80rem;
  margin-left: 0.3rem;
  margin-right: 0.3rem;
  text-decoration: line-through;
  text-align: center;
}

.product-card-colors {
  display: block;
  margin-top: 0.8rem;
}

.product-card-color-option {
  display: inline-block;
  height: 25px;
  width: 25px;
}

.thumbnail{
  flex-basis: 30%;
}

