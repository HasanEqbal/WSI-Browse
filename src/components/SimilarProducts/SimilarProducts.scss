.expanded {
  margin-top: 10px !important;
}



.product-description {
  flex: 1;
  height: auto;
  display: flex;
  align-items: flex-end;
}

.button.tiny {
  margin-left: 11%;
}

.similar-products-section.button{
  margin-left: 11%;
}

.similar-products-section.large-up-1{
  margin: 10px !important;
}

h5{
  margin-top: 1rem !important;
  color: #082a4c !important;
}
