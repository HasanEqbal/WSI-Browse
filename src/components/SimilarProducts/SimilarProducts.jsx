import React from 'react';
import ProductThumbnail from '../ProductThumbnail/ProductThumbnail';

export default function SimilarProducts({ productsList }) {
    let similarProducts = productsList.groups

    return (
        <div className="similar-products-section">
            <div className="grid-x grid-padding-x small-up-1 medium-up-1 large-up-1">
                <h5 className="subheader primary">Similar Products</h5>
            </div>
            <div className="grid-x grid-padding-x small-up-1 medium-up-3 large-up-4">
                {similarProducts ? similarProducts.filter((item, index) => index < 4).map((item, index) => {
                    return <ProductThumbnail key={index} product={item} />
                }) : <div>Similar Product not found</div>}
            </div>
        </div>
    )
}
