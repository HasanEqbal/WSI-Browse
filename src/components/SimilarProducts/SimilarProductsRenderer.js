import { connect } from 'react-redux';
import { lifecycle, compose } from 'recompose';
import { getProducts } from '../../actions';
import SimilarProducts from './SimilarProducts';

const mapStateToProps = (state) => {
  return { productsList: state.productsList };
};

const reduxConnector = connect(mapStateToProps, { getProducts });

const lifecycleHandlers = lifecycle({
  componentDidMount() {
    this.props.getProducts();
  }
});

export default compose(
  reduxConnector,
  lifecycleHandlers,
)(SimilarProducts);
