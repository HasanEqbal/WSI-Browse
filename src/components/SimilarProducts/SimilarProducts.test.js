import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import SimilarProducts from './SimilarProducts';
import ProductThumbnail from '../../components/ProductThumbnail/ProductThumbnail';


const fixture = require('../../testData/testFixture.json');

describe('Component : SimilarProduct', () => {

  it('expect to render Similar Product component', () => {
    expect(shallow(<SimilarProducts productsList={fixture} />)).toMatchSnapshot();
  })

  it('verifying rendering of productThumbnail component in Similar Product component', () => {
    const groups = fixture.groups
    const wrapper = mount(
      <MemoryRouter>
        <SimilarProducts productsList={fixture} />
      </MemoryRouter>
    )
    expect(wrapper.contains(<ProductThumbnail product={groups[0]} />)).toBe(true);
    expect(wrapper.contains(<ProductThumbnail product={groups[3]} />)).toBe(true);
    expect(wrapper.contains(<ProductThumbnail product={groups[4]} />)).toBe(false);
  })
})
