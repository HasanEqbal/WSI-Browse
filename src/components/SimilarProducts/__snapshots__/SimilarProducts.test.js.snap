// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component : SimilarProduct expect to render Similar Product component 1`] = `ShallowWrapper {}`;
